src/mcp_agent/workflows/llm/multipart_converter_openai.py

```
@@ -0,0 +1,500 @@
+                elif is_resource_content(item):
+                    block = OpenAIConverter._convert_embedded_resource(item)
+                    if block:
+                        content_blocks.append(block)
+
+                else:
+                    _logger.warning(f"Unsupported content type: {type(item)}")
+                    # Create a text block with information about the skipped content
+                    fallback_text = f"[Unsupported content type: {type(item).__name__}]"
+                    content_blocks.append({"type": "text", "text": fallback_text})
+
+            except Exception as e:
```

Pylint (3.3.7)

[warning] Catching too general exception Exception

(W0718)